prow/gerrit/client/client.go

```diff
@@ -758,13 +758,19 @@ func (h *gerritInstanceHandler) queryChangesForProjectWithoutMetrics(log logrus.
 				if !newMessages {
 					log.Debug("Found updated change")
 				}
-				pending = append(pending, change)
+				pendingChanges[change.Number] = change
 			default:
 				// change has been abandoned, do nothing
 				log.Debug("Ignored change")
 			}
 		}
 	}
+return_result:
+	result := []gerrit.ChangeInfo{}
+	for _, change := range pendingChanges {
+		result = append(result, change)
+	}
```

Comment:
This loop does not preserve the order of the changes, but I believe you indicated that would not be an issue.

Reply:
@cjwagner said offline that the API does return things in chronological order, so we want to preserve it. PR has been updated with your patch to preserve order.